lib/request.js

```
@@ -0,0 +1,54 @@
+var client = {
+    https: require('https'),
+    http: require('http'),
+};
+
+var request = function request(options) {
+    options.simple = options.simple !== false;
+    function promiseRequest(resolve, reject) {
+        if (options.json) {
+            options.headers['Content-Type'] = 'application/json';
+        }
+
+        var req = client[options.uri.split(':')[0]].request(options.uri, options, (res)=>{
```

Should throw a clean error if protocol isn't supported

---

why it should be a syntax error since your not using it for a http request?